### ISSUE TITLE
How do you get the actual rank (in position)? 

### ISSUE THREAD

**Issue by creativereason** [mixonic/ranked-model#116]
I am using and like using `ranked_model` for easily sorting data when it's added (and the easy add-on for AJAX). I've used it on a few different projects. But either I need my coffee right now or I'm missing this from the documentation. 

Let's say I have a list of `pages` in an `Issue` using `position` to assign them . 
```
class Page < ActiveRecord::Base
...
  include RankedModel
    ranks :position,
    :with_same => :issue_id    # Pages belong_to Issues, make the ranker scoped to one issue

```
How would I find out which page (number) is a given page is in issue? Is there a way to get an index of that using RankedModel?

**Comment by mixonic** [comment]
Sorry, can you restate this?

> How would I find out which page (number) is a given page is in issue?

It is a bit hard to parse, perhaps you need some ☕️ first ;-)

**Comment by creativereason** [comment]
Sorry definitely need coffee ;-). 

If I have 20 pages in ranked_model how do I figure out the index of one of them. So even though my 3rd item is 2145386478 (using ranked_model's position). Is there any way to know it's 3rd in the list (or second in a zero-based array)? I feel like I've done this before but don't see it in documentation.

**Comment by mixonic** [comment]
I suggest we use the term "position" to mean the place of a record in the ordered sequence (1, 2, 3) and "rank" to mean the integer representation in a sparse space (-54543, 30, 302423, 5023423). These should be the terms generally used in this code and the readme.

I don't believe there is a built-in method for getting the position of a record. You can [fetch a record at a position](https://github.com/mixonic/ranked-model/blob/5fd725bb07a331e0fb10ba3f1b0bee15dad8a47f/spec/duck-model/lots_of_ducks_spec.rb#L21), but cannot get a position from a record.

With a few scopes and a `count` statement you could find the number of records with a lower rank than the record you have. That would give you the position. Something like:

```ruby
# Given `howard` as a `Duck` instance
howard_position = Duck.where(Duck.arel_table[:age].lt(howard.age)).count
```

**Comment by creativereason** [comment]
Ok, thanks for the clarification. Is this functionality (adding position, in addition to rank) something you'd think would benefit the gem?

**Comment by mixonic** [comment]
I'm unsure- It is the kind of thing you may want some amount of friction to exist around. For example you would not want to look through a set of records (say for a JSON api serialization) and find their positions- it would involve a lot of DB hits and be a performance hit.

In general the design of this system is to bend the performance tradeoffs normally present- It is expensive to find the position of a record, but very cheap to add a record or delete one.

**Comment by creativereason** [comment]
Makes sense to me. In case this helps anyone in the future, I did something like this. In my case I don't have more than 100 objects per ranked_model (and they are filtered by the a parent_id, so it's really a small array). 

```
class Duck < ActiveRecord::Base
  include RankedModel
    ranks :rank

  def position
    ducks = Duck.all 
    ducks.map(&:id).index(self.id)
  end

```
